homeassistant/components/media_player/onkyo_serial.py

```diff
@@ -0,0 +1,198 @@
+"""
+Support for Onkyo Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.onkyoserial/
+"""
+import logging
+import voluptuous as vol
+from yaml import load
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_SELECT_SOURCE, SUPPORT_PLAY, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (STATE_OFF, STATE_ON, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = [
+    'pyserial>=3.1.2',
+    'https://github.com/rayzorben/onkyo-serial/archive/0.0.2.zip'
+    '#onkyo_serial>=0.0.2'
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_ONKYO = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE | SUPPORT_PLAY
+
+CONF_SOURCES = 'sources'
+CONF_ZONES = 'zones'
+CONF_PORT = 'port'
+
+DEFAULT_NAME = 'Onkyo Serial Receiver'
+DEFAULT_PORT = '/dev/ttyUSB0'
+DEFAULT_ZONES = load("""

```

Comment:
changed, since i am providing default yaml for the configuration, my method seems more consistent and readable, but updated anyway.